### ISSUE TITLE
Rule proposal: Forbid `g` flag used on `RegExp#test()` 

### ISSUE THREAD

**Issue by fisker** [ota-meshi/eslint-plugin-regexp#118]
If the regex is only used once on `RegExp#test()`, the `g` flag is useless.

#### Fail

```js
/foo/g.test(bar);
```

```js
new RegExp('foo', 'g').test(bar);
```

#### Pass

```js
const regex = /foo/g;
regex.test(bar);
```

I original make this proposal [here](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1174), but feel better put `RegExp` related rules together in this codebase.

**Comment by ota-meshi** [comment]
Thank you for proposing this rule! This rule is sounds good to me!

Do you think the rule should also be reported for the next usage?

```js
"foofoo".search(/foo/g)
```

**Comment by fisker** [comment]
Yes, nice catch.

**Comment by fisker** [comment]
It will be great if can also trace variables and report the one only used once on `test`/`search`, but not sure if it worth the effort.
